chaincode/ledger.go

```diff
@@ -88,57 +88,72 @@ type AggregateAlgo struct {
+// that are common to Traintuple, CompositeTraintuple and
+// AggregateTuple
+type GenericTuple struct {
+	AssetType     AssetType `json:"assetType"`
+	ObjectiveKey  string    `json:"objectiveKey"`
+	AlgoKey       string    `json:"algoKey"`
+	ComputePlanID string    `json:"computePlanID"`
+	Creator       string    `json:"creator"`
+	Log           string    `json:"log"`
+	Rank          int       `json:"rank"`
+	Status        string    `json:"status"`
+	Tag           string    `json:"tag"`
+}
+
 // Traintuple is the representation of one the element type stored in the ledger. It describes a training task occuring on the platform
 type Traintuple struct {
 	AssetType     AssetType   `json:"assetType"`
+	ObjectiveKey  string      `json:"objectiveKey"`
 	AlgoKey       string      `json:"algoKey"`
+	ComputePlanID string      `json:"computePlanID"`
 	Creator       string      `json:"creator"`
+	Log           string      `json:"log"`
+	Rank          int         `json:"rank"`
+	Status        string      `json:"status"`
+	Tag           string      `json:"tag"`
 	Dataset       *Dataset    `json:"dataset"`
-	ComputePlanID string      `json:"computePlanID"`
 	InModelKeys   []string    `json:"inModels"`
-	Log           string      `json:"log"`
-	ObjectiveKey  string      `json:"objectiveKey"`
 	OutModel      *HashDress  `json:"outModel"`
 	Perf          float32     `json:"perf"`
 	Permissions   Permissions `json:"permissions"`
-	Rank          int         `json:"rank"`
-	Status        string      `json:"status"`
-	Tag           string      `json:"tag"`
 }
 
 // CompositeTraintuple is like a traintuple, but for composite model composition
 type CompositeTraintuple struct {
-	AlgoKey       string                      `json:"algoKey"`
 	AssetType     AssetType                   `json:"assetType"`
+	ObjectiveKey  string                      `json:"objectiveKey"`
+	AlgoKey       string                      `json:"algoKey"`
 	ComputePlanID string                      `json:"computePlanID"`
 	Creator       string                      `json:"creator"`
+	Log           string                      `json:"log"`
+	Rank          int                         `json:"rank"`
+	Status        string                      `json:"status"`
+	Tag           string                      `json:"tag"`
 	Dataset       *Dataset                    `json:"dataset"`
 	InHeadModel   string                      `json:"inHeadModel"`
 	InTrunkModel  string                      `json:"inTrunkModel"`
-	Log           string                      `json:"log"`
-	ObjectiveKey  string                      `json:"objectiveKey"`
 	OutHeadModel  CompositeTraintupleOutModel `json:"outHeadModel"`
 	OutTrunkModel CompositeTraintupleOutModel `json:"outTrunkModel"`
 	Perf          float32                     `json:"perf"`
-	Rank          int                         `json:"rank"`
-	Status        string                      `json:"status"`
-	Tag           string                      `json:"tag"`
```

Comment:
I don't mind resorting struct element but I'm not a sure what's the logic behind this order (also an alphabetic order was ok for me)

Reply:
I sorted it this way so that the common members between Traintuple, CompositeTraintuple and AggregateTuple are all at the top (and hopefully can eventually be removed altogether, in favor of structure composition where `GenericTuple` would be included as part of each sub-structure? to be discussed :p)

I can revert to the original sort, if you prefer?

Reply:
discussed with @thibaultrobert : no strong opinion, let's leave it as it is now.